src/amp-story-player/amp-story-component-manager.js

```diff
@@ -87,7 +119,12 @@ export class AmpStoryComponentManager {
    */
   layoutIfVisible_(elImpl) {
     const elTop = elImpl.getElement()./*OK*/ getBoundingClientRect().top;
-    if (this.win_./*OK*/ innerHeight * 2 > elTop) {
+    const winInnerHeight = this.win_./*OK*/ innerHeight;
+
+    if (winInnerHeight * MIN_VIEWPORT_PRERENDER_DISTANCE > elTop) {
+      elImpl.prerenderCallback();
```

Comment:
Same thing, do you know if this would get called just once? (just curious)

Reply:
It would normally trigger whenever the criteria of the `IO` is met (i.e. whatever is described on its `rootMargin` or `threshold` in its init parameters). But we are also calling `IO.unobserve()` once it triggers the first time, which stops observing the element and thus won't trigger a second time.

We also have the `if (this.isLaidOut_) return;` in the player code to stop from running a second time in case the fallback for the `IO` is needed in unsupported browsers.